examples/abbas_work_area/OOP_contact.py

```diff
@@ -0,0 +1,152 @@
+import json
+import os
+from OOP_contact_record import ContactRecord
+from CSVdatastore import CSVDataStore
+from TXTdatastore import TXTDataStore
+from JSONdatastore import JSONDataStore
+from examples.abbas_work_area.datastore import DataStore
+
+
+class ContactBook:
+    BASE_FILE_PATH = "C:/Users/71519/Documents/genz-todo-list-app/examples/abbas_work_area/data/store_contacts"
+
+    def __init__(self, storage_format='csv'):
+        self.set_datastore(storage_format)
+        self.storage_format = storage_format
+        self.datastore: DataStore = CSVDataStore()
+        self.CONTACT_FILE_PATH = f"{self.BASE_FILE_PATH}.{storage_format}"
+
+    def set_datastore(self, storage_format):
+        if storage_format == 'csv':
+            self.datastore = CSVDataStore()
+        elif storage_format == 'txt':
+            self.datastore = TXTDataStore()
+        elif storage_format == 'json':
+            self.datastore = JSONDataStore()
+        self.CONTACT_FILE_PATH = f"{self.BASE_FILE_PATH}.{storage_format}"
+
+    def write_heading(self):
+        self.open_write_file("Name,Contact,Email,Address\n", 'w')  # writes heading of csv
+
+    def if_header_exists(self):
+        if not os.path.exists(self.CONTACT_FILE_PATH):  # checks if file exists
+            self.write_heading()
+
+    def open_write_file(self, data="", state="r"):
+        """
+        contains all file related functions such as, Write, Read, Append to main file
+        :param data: user inputted record
+        :param state: a, r, or w
+        :return: Data Saved Successfully!
+        """
+        with open(self.CONTACT_FILE_PATH, state) as contact_file:
+            if state in ["a", "w"]:
+                contact_file.write(data)
+            elif state == "r":
+                return [x.strip() for x in contact_file.readlines()]
+
+    # chooses how to format the data based on users choice and chooses the correct class
+    def format_record(self, name, contact, email, address):
+        """
+        :param name: name
+        :param contact: phone number
+        :param email: email
+        :param address: address
+        :return: chooses the format of file based on user input
+        """
+        contact_record = ContactRecord(name, contact, email, address)
+        return self.datastore.format(contact_record)
+
+    def store_contact(self, name, contact, email, address):
+        contact_record = ContactRecord(name, contact, email, address)
+        store = CSVDataStore()
+        record_str = store.format(contact_record)
```

Comment:
Based on the implementation, we can define separate DataFormatter class and use it in specific DataStore classes